drivers/serial/CMakeLists.txt

```diff
@@ -26,7 +26,13 @@ zephyr_library_sources_ifdef(CONFIG_UART_MIV uart_miv.c)
 zephyr_library_sources_ifdef(CONFIG_UART_MSP432P4XX uart_msp432p4xx.c)
 zephyr_library_sources_ifdef(CONFIG_UART_NS16550 uart_ns16550.c)
 zephyr_library_sources_ifdef(CONFIG_UART_NRFX_UART uart_nrfx_uart.c)
-zephyr_library_sources_ifdef(CONFIG_UART_NRFX_UARTE uart_nrfx_uarte.c)
+if (CONFIG_UART_NRFX_UARTE)
+  if (CONFIG_UART_NRFX_UARTE_LEGACY_SHIM)
+    zephyr_library_sources(uart_nrfx_uarte.c)
+  else()
+    zephyr_library_sources(uart_nrfx_uarte2.c)
```

Comment:
Not really a fan of having "uarte.c" and "uarte2.c" isn't there a better way to distinguish them?

Reply:
if there are commits which shouldn't be reviewed in this PR because they are part of another, dependent PR, then please squash ALL such commits, and give the squashed commit a message, for example like this:
```
squashed commits from PR(s) #xxxxx #yyyyy etc.

This commit should not be reviewed in this PR.
```

That will make it much easier for reviewers to skip the commits which should not be reviewed in this PR.